purchases/src/customEntitlementComputation/kotlin/com/revenuecat/purchases/PurchasesConfiguration.kt

```diff
@@ -0,0 +1,105 @@
+package com.revenuecat.purchases
+
+import android.content.Context
+import java.util.concurrent.ExecutorService
+
+open class PurchasesConfiguration(builder: Builder) {
+
+    val context: Context
+    val apiKey: String
+    val appUserID: String
+    val observerMode: Boolean
+    val service: ExecutorService?
+    val store: Store
+    val diagnosticsEnabled: Boolean
+    val dangerousSettings: DangerousSettings
+    val verificationMode: EntitlementVerificationMode
+
+    init {
+        this.context = builder.context
+        this.apiKey = builder.apiKey
+        this.appUserID = builder.appUserID
+        this.observerMode = builder.observerMode
+        this.service = builder.service
+        this.store = builder.store
+        this.diagnosticsEnabled = builder.diagnosticsEnabled
+        this.verificationMode = builder.verificationMode
+        this.dangerousSettings = builder.dangerousSettings
+    }
+
+    open class Builder(
+        @get:JvmSynthetic internal val context: Context,
+        @get:JvmSynthetic internal val apiKey: String,
+        @get:JvmSynthetic internal val appUserID: String,
+    ) {
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var observerMode: Boolean = false
+
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var service: ExecutorService? = null
+
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var store: Store = Store.PLAY_STORE
+
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var diagnosticsEnabled: Boolean = false
+
+        @set:JvmSynthetic @get:JvmSynthetic
+        internal var verificationMode: EntitlementVerificationMode = EntitlementVerificationMode.default
+
+        // TODO Default to custom entitlement computation mode
```

Comment:
The dangerous setting is not on this branch. To avoid some conflicts, I'm just keeping it as a TODO for now, and once this is rebased, I will add it